src/scrapers/isracard.ts

```diff
@@ -0,0 +1,236 @@
+import * as puppeteer from 'puppeteer';
+import * as lodash from 'lodash';
+import { fetchGetWithinPage, fetchPostWithinPage } from '../utils/fetch';
+import { IsracardDashboardMonth } from '../../generatedTypes/isracardDashboardMonth';
+import { IsracardCardsTransactionsList } from '../../generatedTypes/isracardCardsTransactionsList';
+import * as isracardDashboardMonth from '../schemas/isracardDashboardMonth.json';
+import * as isracardCardsTransactionsList from '../schemas/isracardCardsTransactionsList.json';
+import { validateSchema } from '../utils/validateSchema';
+
+const SERVICE_URL =
+  'https://digital.isracard.co.il/services/ProxyRequestHandler.ashx';
+
+async function login(credentials: isracardCredentials, page: puppeteer.Page) {
+  const validateUrl = `${SERVICE_URL}?reqName=performLogonI`;
+  const validateRequest = {
+    MisparZihuy: credentials.ID,
+    Sisma: credentials.password,
+    cardSuffix: credentials.card6Digits,
+    countryCode: '212',
+    idType: '1',
+  };
+  return fetchPostWithinPage(page, validateUrl, validateRequest);
+}
+
+async function fetchAndEditMonth(
+  page: puppeteer.Page,
+  monthDate: Date,
+  options?: isracardOptions
+) {
+  // get accounts data
+  const billingDate = `${monthDate.getFullYear()}-${("0"+(monthDate.getMonth()+1)).slice(-2)}-${("0" + monthDate.getDate()).slice(-2)}`; // get date in format YYYY-MM-DD
```

Comment:
It's cleaner, saw it on my search when created this not-so-elegant solution.
But it creates string in the wrong format, I'll check if with this can be solved.